examples/modules/example_module.cpp

```diff
@@ -65,27 +65,24 @@ class MyModule : public GenericService::Service, public ComponentBase {
 };
 
 int MyModule::which = 0;
-std::shared_ptr<ModuleService_> my_mod;
-
-void signal_handler(int signum) {
-    my_mod->close();
-}
 
 int main(int argc, char** argv) {
     if (argc != 2) {
         throw "need socket path as command line argument";
     }
 
-    // TODO(RSDK-1920) This is still causing non-graceful shutdown. Figure out why, and fix.
-    struct sigaction sig_handler;
-    sig_handler.sa_handler = signal_handler;
-    sigaction(SIGTERM, &sig_handler, nullptr);
-    sigemptyset(&sig_handler.sa_mask);
-    sig_handler.sa_flags = 0;
+    // C++ modules must handle SIGINT and SIGTERM. Make sure to create a sigset
+    // for SIGINT and SIGTERM that can be later awaited in a thread that cleanly
+    // shuts down your module.
+    sigset_t sigset;
```

Comment:
If you wanted to reduce the amount of boilerplate in the example, you could create a static method on `Server` or `ModuleService` to mask and return a sigset appropriate for the signals the module must handle.

Reply:
My worry about helper functions for sigset creation and signal-checking-thread creation is that it will make this logic harder to read without significantly reducing the amount of code here. If a user is building any sort of complex module, I have a feeling they'll want to do their own custom signal handling and capture (potentially shutting down custom child processes or destroying custom objects). It might be best to show explicitly how to handle signals entirely within the module and let the user either copy that code exactly or modify it to what they need. What are your thoughts?

Reply:
I think having more boilerplate is fine - or even desirable - in an example. Would rather have slightly wordier code here that spells out explicitly how to do signal handling.

Reply:
Sounds good to me.